src/boards/mikoto/board.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Hai-Ninh Dang
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef _MIKOTO_H
+#define _MIKOTO_H
+
+#define _PINNUM(port, pin)    ((port)*32 + (pin))
+
+#define UICR_REGOUT0_VALUE UICR_REGOUT0_VOUT_3V3
+
+/*------------------------------------------------------------------*/
+/* LED
+ *------------------------------------------------------------------*/
+#define LEDS_NUMBER       1
+#define LED_PRIMARY_PIN   _PINNUM(0, 15) // Blue
+#define LED_STATE_ON      1
+
+/*------------------------------------------------------------------*/
+/* BUTTON
+ *------------------------------------------------------------------*/
+#define BUTTONS_NUMBER    2  // none connected at all
+#define BUTTON_1          _PINNUM(0, 18)  // unusable: RESET
+#define BUTTON_2          _PINNUM(0, 19)  // no connection
+#define BUTTON_PULL       NRF_GPIO_PIN_PULLUP
+
+//--------------------------------------------------------------------+
+// BLE OTA
+//--------------------------------------------------------------------+
+#define BLEDIS_MANUFACTURER  "zhiayang"
+#define BLEDIS_MODEL         "mikoto"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID           0x1209
+#define USB_DESC_UF2_PID       0x7A01
+#define USB_DESC_CDC_ONLY_PID  0x7A01
+
+#define UF2_PRODUCT_NAME  "Mikoto nRF52840"
+#define UF2_VOLUME_LABEL  "MIKOTO-BOOT"
+#define UF2_BOARD_ID      "nRF52840-mikoto"
```

Comment:
I am not picky but board id must follow uf2 spec https://github.com/microsoft/uf2#files-exposed-by-bootloaders . You need to add the board revision as well

```
The Board-ID field is machine-readable and consists of a number of dash-separated tokens. The first token is the CPU type, second is the board type, and third is the board revision. More tokens can be also added.
```

Reply:
> 

Thank you, I changed that.